### PR TITLE
Add check for subtasks before setSubTaskVisibility

### PR DESCRIPTION
**Pull request by mcbouslog**
Package: lib-classifier

Fixes subtask showing after all drawing tools, even if no subtasks.

Describe your changes:
- adds conditional check for subtasks (`mark.tasks`) before setting subtask visibility (`setSubTaskVisibility`)

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by mcbouslog** [comment]
Alternatively or additionally, a conditional could be added in the [DrawingTask model here](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js#L83) to check if the `self.activeMark` has a `tasks` length?

**Comment by mcbouslog** [comment]
`onPointerDown` and `onDelete` also call `setSubTaskVisibility` with `false`. Not sure if conditional should be added to those as well, not a problem as is, but maybe best to add there as well, if this all stays in `InteractionLayer` as opposed to exclusively in `DrawingTask`.

**Comment by shaunanoordin** [comment]
Yikes! I can't believe I forgot to test SubTask visibility for Drawing Tasks that don't have sub-tasks. Good catch!

Regarding your comments, I too was thinking that the `setSubTaskVisibility()`'s internals should be responsible for checking if it _should_ set visibility to true, but I don't feel _too_ strongly about it.

e.g. I think this is good:
```
function setSubTaskVisibility (visible, drawingMarkNode = undefined, subtasks = []) {
  // Only show the subtask popup IF asked to be visible AND there are subtasks available 
  self.subTaskVisibility = (visible && subtasks?.length > 0)
  
  self.subTaskMarkBounds = (drawingMarkNode)
    ? drawingMarkNode.getBoundingClientRect()
    : undefined
}
```

That makes the logic more _self-contained_ in one function, yay, but it does make that one function look a bit messy, boo (that's a lot of variables to pass into one function).

It's fine either way, but if you're keeping your code the way it is, I'd recommend...

```
if (activeMark && activeMark.tasks?.length > 0) {
// -->
if (activeMark?.tasks?.length > 0) {

// ...and...
if (mark.tasks?.length > 0) {
// -->
if (mark?.tasks?.length > 0) {
```

File: packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js [hunk 1]
```diff
@@ -81,10 +81,12 @@ export const Drawing = types.model('Drawing', {
     }
 
     function setSubTaskVisibility (visible, drawingMarkNode) {
-      self.subTaskVisibility = visible
-      self.subTaskMarkBounds = (drawingMarkNode)
-        ? drawingMarkNode.getBoundingClientRect()
-        : undefined
+      if (self.activeTool.tasks?.length > 0) {
```
[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js]: Yes great catch!

**Comment by eatyourgreens** [comment]
I think you get the first tool back from the drawing task if one hasn’t been actively selected. The first one is selected by default in the task panel.

**Comment by eatyourgreens** [comment]
Oh wait, we allow you to build drawing tasks without any tools?

**Pull request merged**